### ISSUE TITLE
Unable to reproduce the results?

### ISSUE THREAD

**Issue by dingyf05** [suduo94/D4M#5]
Hi, I tried to reproduce the results following the steps in readme. The generated images look good and realistic to me, but I only got a top-1 err near 90 under the resnet18 and 10ipc setting (which should be 100 - 27.9 = 72.1 in paper?).  The only difference is that I'm training on 1 single A100 gpu instead of 8 (same batchsize, same learning rate) but I guess that should make no difference?

I'm wondering is it possible to also provide a env/docker file, or maybe the generated dataset and the features for kl loss? I'm using the latest pytorch/diffusers library. Since your project requires us to directly modify the source code of the above library, I'm not sure if I have done it correctly.

My training log:
```
Epoch: 196
TRAIN Iter 196: lr = 0.000001,  loss = 0.001393,        Top-1 err = 99.650000,  Top-5 err = 98.910000,  train_time = 36.656602

Epoch: 197
TRAIN Iter 197: lr = 0.000001,  loss = 0.001418,        Top-1 err = 99.740000,  Top-5 err = 98.930000,  train_time = 35.769536

Epoch: 198
^@TRAIN Iter 198: lr = 0.000000,        loss = 0.001397,        Top-1 err = 99.690000,  Top-5 err = 99.030000,    train_time = 36.054903

Epoch: 199
TRAIN Iter 199: lr = 0.000000,  loss = 0.001414,        Top-1 err = 99.700000,  Top-5 err = 99.040000,  train_time = 36.111678
TEST Iter 199: loss = 3.881970, Top-1 err = 81.090000,  Top-5 err = 55.930000,  val_time = 70.990983

```

**Comment by suduo94** [comment]
Hi there, thanks for your interest.
Could you tell me which dataset you are testing for? Note that our method needs to train with soft labels and you should pre-generate them using FKD or other distillation methods before implementing the classification tasks.

**Comment by dingyf05** [comment]
Hi, I'm using the imagenet1k dataset from the official [pytorch repo](https://github.com/pytorch/examples/tree/main/imagenet)

And yes I'm following the exact three steps in the repo, namely,  distillation(cluster and generation 10ipc), then ttm (generate the corresponding logits from the syn images), and validate(kld training). Just to make sure, only kld loss is used in the validation step (no ce loss), right?.

**Comment by suduo94** [comment]
It should be clear that the validation also includes the training and testing process. The training loss is KL divergence (it must be agreed with the matching process) and the testing loss is CrossEntropy.

**Comment by suduo94** [comment]
We use PyTorch 1.12.1 and diffusers 0.21.0.dev0 in our experiments.
And there exists some inconsistency because the DD task has no fine-grained information as prompts indeed.
You could try to learn more prototypes and choose the best one.

**Comment by haruishi43** [comment]
@dingyf05 I'm also having trouble reproducing the results on imagenet. Did you resolve it?

**Comment by haruishi43** [comment]
@suduo94 thank you for opensourcing your work. For IPC10 on ImageNet, I could only obtain top1-error of 89.126% after 300 epochs (batch size is the same, but I used two gpus). Are there any tips on reproducing your work? Also if it's possible, can you upload your distilled datasets for imagenet?

**Comment by suduo94** [comment]
Hi there, did you use the FKD to generate the soft label when training the downstream tasks? And what are the teacher and student models you used?

**Comment by haruishi43** [comment]
Thanks for the reply. I did use FKD for training. And I also used the default ResNet18 as teacher and student. Besides the number of gpu, I didn't change the hyperparameters. Besides that, I noticed that safety checker for StableDiffusion detected some NSFW images so I disable the safety checker altogether

**Comment by xqx456** [comment]
> Thanks for the reply. I did use FKD for training. And I also used the default ResNet18 as teacher and student. Besides the number of gpu, I didn't change the hyperparameters. Besides that, I noticed that safety checker for StableDiffusion detected some NSFW images so I disable the safety checker altogether

Hello, I still have some questions to ask you. When I ran sh train_FKD.sh for verification, I found an error: raise ValueError('config is not loaded'). Then I checked the code and found that the load_batch_config function was not called to update self.batch_config, which caused self.batch_config to be None. I have adjusted the code for several days but still cannot solve this problem. So I would like to ask if you know the answer. I would be grateful if you can solve this problem. Sorry for the inconvenience.
